autoload/phpactor/input/list.vim

```diff
@@ -0,0 +1,40 @@
+function! phpactor#input#list#strategy()
+    if !has_key(g:, 'phpactor#input#list#strategy')
+        let g:phpactor#input#list#strategy = s:auto_detect_strategy()
```

Comment:
can we add this setting to the plugin doc? https://github.com/phpactor/phpactor/blob/develop/doc/vim-plugin.md#configuration 

Would it make more sense to have `phpactorInputListStrategy` as the global name to be consistent (more consistent) with the existing options?

Reply:
I will try to do the doc tomorrow.
For the name, I had the same idea but I wanted to keep it as close as the autoload functions as possible.
But since the configuration is not autoloaded it makes more sense.

Reply:
Done & Done

I also added a custom error message in case of invalid strategy.
And gave the possibility to provide a custom strategy if a user want to create it's own.